src/context/filtercontext.js

```diff
@@ -1,19 +1,21 @@
 import { createContext, useContext, useReducer } from "react";
 import { filterReducer } from "../reducer/filterreducer";
-const filter = createContext();
-const useFilter = () => useContext(filter);
+const FilterContext = createContext();
 
 const FilterProvider = ({ children }) => {
   const [filterState, filterFunc] = useReducer(filterReducer, {
```

Comment:
make constant of initialState and pass in useReducer it will be clean and scalable